src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/FormCollectionModelBinderProvider.cs

```diff
@@ -19,7 +21,18 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (context.Metadata.ModelType == typeof(IFormCollection))
+            var modelType = context.Metadata.ModelType;
+
+            if (typeof(FormCollection).GetTypeInfo().IsAssignableFrom(modelType))
+            {
+                throw new InvalidOperationException(
```

Comment:
This is different than what the bug suggests at the solution. https://github.com/aspnet/Mvc/issues/4895#issuecomment-227776872

This special cases `FormCollection` - the suggestion was to check for a no-arg constructor in the Complex type provider and instead of throwing an exception, just let it pass through, the system will throw if a model binder can't be created.